src/ILCompiler.Compiler/src/IL/Stubs/StartupCode/NativeLibraryStartupMethod.cs

```diff
@@ -0,0 +1,109 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+
+using Internal.TypeSystem;
+
+using AssemblyName = System.Reflection.AssemblyName;
+using Debug = System.Diagnostics.Debug;
+
+namespace Internal.IL.Stubs.StartupCode
+{
+    /// <summary>
+    /// Startup code that does initialization, Main invocation
+    /// and shutdown of the runtime.
+    /// </summary>
+    public sealed partial class NativeLibraryStartupMethod : ILStubMethod
+    {
+        private TypeDesc _owningType;
+        private MethodSignature _signature;
+        private IList<MethodDesc> _libraryInitializers;
+
+        public NativeLibraryStartupMethod(TypeDesc owningType, IList<MethodDesc> libraryInitializers)
+        {
+            _owningType = owningType;
+            _libraryInitializers = libraryInitializers;
+        }
+
+        protected override int ClassCode => -304225482;
```

Comment:
We put `ClassCode` and `CompareToImpl` in separate dot files so that once we stand up a JIT (that doesn't need to sort) we can compile the JIT codebase without support for this. Granted, it's unlikely we would use this particular class in the JIT codebase, but we've been doing it everywhere for consistency.